### ISSUE TITLE
[`naming-convention`] `objectLiteralProperty` is not working properly

### ISSUE THREAD

**Issue by dalisoft** [typescript-eslint/typescript-eslint#4749]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

**Repro**

https://github.com/dalisoft/airlight/blob/402156ce40c57ac0e85dbc5fb5ca7371fc2587e8/packages/eslint-config-base/config.js

```TS
The configuration itself
```

**Expected Result**

No errors as i'm defined rules for `objectLiteralProperty`

**Actual Result**

Error even on `objectLiteralProperty`

**Additional Info**

<img width="1043" alt="Bug in Screenshot" src="https://user-images.githubusercontent.com/3511344/160363928-eedb50b2-8e60-49b2-a9a7-d8237f62601c.png">

<details>
<summary>Debug logs</summary>

```
yarn run v1.22.18
$ /Users/dalisoft/Desktop/Open_Source/airlight/node_modules/.bin/eslint . --debug

/Users/dalisoft/Desktop/Open_Source/airlight/packages/eslint-config-base/config.js
   23:5  error  Object Literal Property name `@typescript-eslint/naming-convention` must match one of the following formats: snake_case, strictCamelCase                    @typescript-eslint/naming-convention
   70:5  error  Object Literal Property name `@typescript-eslint/explicit-function-return-type` must match one of the following formats: snake_case, strictCamelCase        @typescript-eslint/naming-convention
   71:5  error  Object Literal Property name `@typescript-eslint/explicit-module-boundary-types` must match one of the following formats: snake_case, strictCamelCase       @typescript-eslint/naming-convention
   72:5  error  Object Literal Property name `@typescript-eslint/no-shadow` must match one of the following formats: snake_case, strictCamelCase                            @typescript-eslint/naming-convention
   73:5  error  Object Literal Property name `@typescript-eslint/no-empty-function` must match one of the following formats: snake_case, strictCamelCase                    @typescript-eslint/naming-convention
   74:5  error  Object Literal Property name `@typescript-eslint/indent` must match one of the following formats: snake_case, strictCamelCase                               @typescript-eslint/naming-convention
   75:5  error  Object Literal Property name `@typescript-eslint/class-literal-property-style` must match one of the following formats: snake_case, strictCamelCase         @typescript-eslint/naming-convention
   76:5  error  Object Literal Property name `@typescript-eslint/comma-dangle` must match one of the following formats: snake_case, strictCamelCase                         @typescript-eslint/naming-convention
   77:5  error  Object Literal Property name `@typescript-eslint/comma-spacing` must match one of the following formats: snake_case, strictCamelCase                        @typescript-eslint/naming-convention
   81:5  error  Object Literal Property name `@typescript-eslint/no-unused-vars` must match one of the following formats: snake_case, strictCamelCase                       @typescript-eslint/naming-convention
   82:5  error  Object Literal Property name `@typescript-eslint/no-non-null-asserted-optional-chain` must match one of the following formats: snake_case, strictCamelCase  @typescript-eslint/naming-convention
   83:5  error  Object Literal Property name `@typescript-eslint/restrict-plus-operands` must match one of the following formats: snake_case, strictCamelCase               @typescript-eslint/naming-convention
   84:5  error  Object Literal Property name `operator-linebreak` must match one of the following formats: snake_case, strictCamelCase                                      @typescript-eslint/naming-convention
   85:5  error  Object Literal Property name `prefer-template` must match one of the following formats: snake_case, strictCamelCase                                         @typescript-eslint/naming-convention
   86:5  error  Object Literal Property name `filename-rules/match` must match one of the following formats: snake_case, strictCamelCase                                    @typescript-eslint/naming-convention
  105:5  error  Object Literal Property name `max-depth` must match one of the following formats: snake_case, strictCamelCase                                               @typescript-eslint/naming-convention
  106:5  error  Object Literal Property name `max-nested-callbacks` must match one of the following formats: snake_case, strictCamelCase                                    @typescript-eslint/naming-convention
  107:5  error  Object Literal Property name `max-lines-per-function` must match one of the following formats: snake_case, strictCamelCase                                  @typescript-eslint/naming-convention
  112:5  error  Object Literal Property name `max-lines` must match one of the following formats: snake_case, strictCamelCase                                               @typescript-eslint/naming-convention
  116:5  error  Object Literal Property name `no-useless-concat` must match one of the following formats: snake_case, strictCamelCase                                       @typescript-eslint/naming-convention
  117:5  error  Object Literal Property name `no-console` must match one of the following formats: snake_case, strictCamelCase                                              @typescript-eslint/naming-convention
  118:5  error  Object Literal Property name `no-template-curly-in-string` must match one of the following formats: snake_case, strictCamelCase                             @typescript-eslint/naming-convention
  119:5  error  Object Literal Property name `no-underscore-dangle` must match one of the following formats: snake_case, strictCamelCase                                    @typescript-eslint/naming-convention
  120:5  error  Object Literal Property name `no-shadow` must match one of the following formats: snake_case, strictCamelCase                                               @typescript-eslint/naming-convention
  121:5  error  Object Literal Property name `no-empty-function` must match one of the following formats: snake_case, strictCamelCase                                       @typescript-eslint/naming-convention
  123:5  error  Object Literal Property name `import/prefer-default-export` must match one of the following formats: snake_case, strictCamelCase                            @typescript-eslint/naming-convention
  130:9  error  Object Literal Property name `@typescript-eslint/explicit-function-return-type` must match one of the following formats: snake_case, strictCamelCase        @typescript-eslint/naming-convention
  131:9  error  Object Literal Property name `@typescript-eslint/explicit-module-boundary-types` must match one of the following formats: snake_case, strictCamelCase       @typescript-eslint/naming-convention
  132:9  error  Object Literal Property name `@typescript-eslint/await-thenable` must match one of the following formats: snake_case, strictCamelCase                       @typescript-eslint/naming-convention
  133:9  error  Object Literal Property name `@typescript-eslint/indent` must match one of the following formats: snake_case, strictCamelCase                               @typescript-eslint/naming-convention
  143:9  error  Object Literal Property name `@typescript-eslint/quotes` must match one of the following formats: snake_case, strictCamelCase                               @typescript-eslint/naming-convention
  150:9  error  Object Literal Property name `@typescript-eslint/no-shadow` must match one of the following formats: snake_case, strictCamelCase                            @typescript-eslint/naming-convention
  151:9  error  Object Literal Property name `@typescript-eslint/no-empty-function` must match one of the following formats: snake_case, strictCamelCase                    @typescript-eslint/naming-convention
  152:9  error  Object Literal Property name `@typescript-eslint/consistent-indexed-object-style` must match one of the following formats: snake_case, strictCamelCase      @typescript-eslint/naming-convention
  155:9  error  Object Literal Property name `no-shadow` must match one of the following formats: snake_case, strictCamelCase                                               @typescript-eslint/naming-convention
  156:9  error  Object Literal Property name `no-empty-function` must match one of the following formats: snake_case, strictCamelCase                                       @typescript-eslint/naming-convention
  164:5  error  Object Literal Property name `import/resolver` must match one of the following formats: snake_case, strictCamelCase                                         @typescript-eslint/naming-convention
  165:1  error  File has too many lines (168). Maximum allowed is 160                                                                                                       max-lines
  169:5  error  Object Literal Property name `import/extensions` must match one of the following formats: snake_case, strictCamelCase                                       @typescript-eslint/naming-convention
  170:5  error  Object Literal Property name `import/external-module-folders` must match one of the following formats: snake_case, strictCamelCase                          @typescript-eslint/naming-convention

/Users/dalisoft/Desktop/Open_Source/airlight/packages/eslint-config-node/config.js
  20:9  error  Object Literal Property name `import/extensions` must match one of the following formats: snake_case, strictCamelCase                       @typescript-eslint/naming-convention
  34:9  error  Object Literal Property name `@typescript-eslint/no-var-requires` must match one of the following formats: snake_case, strictCamelCase      @typescript-eslint/naming-convention
  40:9  error  Object Literal Property name `node/no-unsupported-features/es-syntax` must match one of the following formats: snake_case, strictCamelCase  @typescript-eslint/naming-convention
  55:5  error  Object Literal Property name `import/resolver` must match one of the following formats: snake_case, strictCamelCase                         @typescript-eslint/naming-convention
  62:5  error  Object Literal Property name `import/extensions` must match one of the following formats: snake_case, strictCamelCase                       @typescript-eslint/naming-convention

/Users/dalisoft/Desktop/Open_Source/airlight/packages/eslint-config-react/config.js
  31:5  error  Object Literal Property name `import/extensions` must match one of the following formats: snake_case, strictCamelCase                     @typescript-eslint/naming-convention
  44:5  error  Object Literal Property name `react/prop-types` must match one of the following formats: snake_case, strictCamelCase                      @typescript-eslint/naming-convention
  54:3  error  Object Literal Property name `react/jsx-pascal-case` must match one of the following formats: snake_case, strictCamelCase                 @typescript-eslint/naming-convention
  55:3  error  Object Literal Property name `react/jsx-no-useless-fragment` must match one of the following formats: snake_case, strictCamelCase         @typescript-eslint/naming-convention
  56:3  error  Object Literal Property name `react/no-deprecated` must match one of the following formats: snake_case, strictCamelCase                   @typescript-eslint/naming-convention
  57:3  error  Object Literal Property name `react/jsx-filename-extension` must match one of the following formats: snake_case, strictCamelCase          @typescript-eslint/naming-convention
  71:3  error  Object Literal Property name `@typescript-eslint/naming-convention` must match one of the following formats: snake_case, strictCamelCase  @typescript-eslint/naming-convention

/Users/dalisoft/Desktop/Open_Source/airlight/packages/husky-config/config.js
  3:5  error  Object Literal Property name `pre-commit` must match one of the following formats: snake_case, strictCamelCase  @typescript-eslint/naming-convention
  4:5  error  Object Literal Property name `commit-msg` must match one of the following formats: snake_case, strictCamelCase  @typescript-eslint/naming-convention

/Users/dalisoft/Desktop/Open_Source/airlight/packages/husky-config/interface.d.ts
  2:3  error  Type Property name `pre-commit` must match one of the following formats: snake_case, strictCamelCase  @typescript-eslint/naming-convention
  3:3  error  Type Property name `commit-msg` must match one of the following formats: snake_case, strictCamelCase  @typescript-eslint/naming-convention

✖ 56 problems (56 errors, 0 warnings)

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

</details>

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `5.16.0` |
| `@typescript-eslint/parser`        | `5.0.0` |
| `TypeScript`                       | `4.6.3` |
| `ESLint`                           | `8.12.0` |
| `node`                             | `17.8.0` |

**Comment by JoshuaKGoldberg** [comment]
@dalisoft I can't reproduce your issue on the latest typescript-eslint version. Could you please post a standalone reproduction case -- something that we can get to demonstrate the reported log on our computers? It should at the very least include your `package.json` so we can see what versions of everything you're using.

**Comment by dalisoft** [comment]
@JoshuaKGoldberg 
Thank you for reply and trying help.
Here you can try it:
- `git clone https://github.com/dalisoft/airlight.git`
- `cd airlight`
- `git checkout bug/naming-convention`
- `yarn lint`

And you can see error there defined `kebab-case` for `objectLiteralProperty`

https://github.com/dalisoft/airlight/blob/bug/naming-convention/packages/eslint-config-base/config.js#L36-L42

It worked before good or this property was not added but after some updates (auto dependabot updates) things got break

**Comment by bradzacher** [comment]
I think you'll find that the error message is wrong, it the error is right.

Neither `@foo` nor `foo/bar` match your custom regex `^([a-z][a-z0-9]*)(-[a-z0-9]+)*$`

But the incorrect error message is a bug.

**Comment by dalisoft** [comment]
@bradzacher @JoshuaKGoldberg Not it is matching based on manual RegExp testing
See https://github.com/dalisoft/airlight/blob/bug/naming-convention/packages/eslint-config-base/rules/naming-convention.js#L15-L29

**Comment by bradzacher** [comment]
Okay looking into this - testing against master.

Your original config should actually throw because it's invalid:
https://github.com/dalisoft/airlight/blob/402156ce40c57ac0e85dbc5fb5ca7371fc2587e8/packages/eslint-config-base/config.js#L35-L42
```
Configuration for rule "naming-convention" is invalid:
        Value {"selector":"objectLiteralProperty","leadingUnderscore":"forbid","custom":{"regex":"^([a-z][a-z0-9]*)(-[a-z0-9]+)*$","match":true}} should have required property 'format'.
        ...
```

To fix this it needs to have `format: null`. When I add that to the config it works correctly and shows the correct error messages everywhere:
```
Object Literal Property name `@typescript-eslint/naming-convention` must match the RegExp: /^([a-z][a-z0-9]*)(-[a-z0-9]+)*$/u
```

Then as mentioned - this error is 100% correct because your regex is incorrect.

---

Moving on to your second example:
https://github.com/dalisoft/airlight/blob/d5b6b0a4407a35f6027bab6e7dbe1219b74282d5/packages/eslint-config-base/rules/naming-convention.js#L14-L31

This example is also invalid because you have specified the same selector twice without any modifier differences. I assume what you meant was to use `filter` instead of `custom`?

Making that change makes it work as expected.

---

All that being said - I don't know why you're even bothering attempting to go this route of configuring the lint rule to satisfy the ESLint config? The ESLint config is a pre-specified file wherein you have ZERO control over the property names.

You should just turn the rule off on the file and move on.
